### PR TITLE
First pass at Look component

### PR DESCRIPTION
**Pull request by kelseyc18**

### CODE REVIEW

File: appinventor/appengine/src/com/google/appinventor/client/editor/simple/palette/SimpleComponentDescriptor.java [hunk 1]
```diff
@@ -10,44 +10,7 @@
...
-import com.google.appinventor.client.editor.simple.components.MockScrollVerticalArrangement;
-import com.google.appinventor.client.editor.simple.components.MockSlider;
-import com.google.appinventor.client.editor.simple.components.MockSpinner;
-import com.google.appinventor.client.editor.simple.components.MockTableArrangement;
-import com.google.appinventor.client.editor.simple.components.MockTextBox;
-import com.google.appinventor.client.editor.simple.components.MockTimePicker;
-import com.google.appinventor.client.editor.simple.components.MockVerticalArrangement;
-import com.google.appinventor.client.editor.simple.components.MockVideoPlayer;
-import com.google.appinventor.client.editor.simple.components.MockWebViewer;
+import com.google.appinventor.client.editor.simple.components.*;
```
Comment (ColinTree) [inline appinventor/appengine/src/com/google/appinventor/client/editor/simple/palette/SimpleComponentDescriptor.java]: We dont prefer this way here.

File: appinventor/components/src/com/google/appinventor/components/runtime/Look.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+import android.util.Base64;
+import android.util.Log;
+import android.view.View;
+import android.webkit.JavascriptInterface;
+import android.webkit.ValueCallback;
+import android.webkit.WebViewClient;
+import android.webkit.WebChromeClient;
+import android.webkit.WebView;
+import android.webkit.PermissionRequest;
+import com.google.appinventor.components.annotations.*;
```
Comment (ColinTree) [inline appinventor/components/src/com/google/appinventor/components/runtime/Look.java]: Same, please import specific classes.

File: appinventor/components/src/com/google/appinventor/components/runtime/Look.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+@SimpleObject
+@UsesAssets(fileNames = "look.html, look.js, group1-shard1of1, group10-shard1of1, group11-shard1of1, group12-shard1of1, group13-shard1of1, group14-shard1of1, group15-shard1of1, group16-shard1of1, group17-shard1of1, group18-shard1of1, group19-shard1of1, group2-shard1of1, group20-shard1of1, group21-shard1of1, group22-shard1of1, group23-shard1of1, group24-shard1of1, group25-shard1of1, group26-shard1of1, group27-shard1of1, group28-shard1of1, group29-shard1of1, group3-shard1of1, group30-shard1of1, group31-shard1of1, group32-shard1of1, group33-shard1of1, group34-shard1of1, group35-shard1of1, group36-shard1of1, group37-shard1of1, group38-shard1of1, group39-shard1of1, group4-shard1of1, group40-shard1of1, group41-shard1of1, group42-shard1of1, group43-shard1of1, group44-shard1of1, group45-shard1of1, group46-shard1of1, group47-shard1of1, group48-shard1of1, group49-shard1of1, group5-shard1of1, group50-shard1of1, group51-shard1of1, group52-shard1of1, group53-shard1of1, group54-shard1of1, group55-shard1of1, group6-shard1of1, group7-shard1of1, group8-shard1of1, group9-shard1of1, imagenet_classes.js, model.json, tfjs-0.6.1.js")
```
Comment (ColinTree) [inline appinventor/components/src/com/google/appinventor/components/runtime/Look.java]: This line is too long.. Split it into lines or is it possible to add all group-share files together?

File: appinventor/components/src/com/google/appinventor/components/runtime/Look.java [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    private final WebView webview;
+    private final Form form;
+    private static TensorflowJSHTTPD httpdServer = null;
+
+    /**
+     * Creates a new WebViewer component.
+     *
+     * @param form the container that this component will be placed in
+     */
+    public Look(Form form) {
```
Comment (ColinTree) [inline appinventor/components/src/com/google/appinventor/components/runtime/Look.java]: Why it is not ComponentContainer here.

**Comment by ewpatton** [comment]
@ColinTree This is a component in development by some of our students here at MIT, and as a work in progress there will need to be some changes made. Please try to be more considerate with first time contributors.

**Comment by halatmit** [comment]
I can build and run this with the phone connected, but I can't build an APK.  When I do that, App Inventor says "preparing application icon" and then "preparing final package".  But then it says "preparing application icon" AGAIN, and hangs there.   I'm guessing there's a bug in the build process, possibly having to do with the assets?

**Comment by ewpatton** [comment]
The Look component is now available as an extension at http://appinventor.mit.edu/extensions/.

**Pull request closed without merging**